app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java

```diff
@@ -956,63 +993,90 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
-                if (currentBrightness >= 1f) currentBrightness = 1f;
-                if (currentBrightness <= minBrightness) currentBrightness = minBrightness;
 
-                lp.screenBrightness = currentBrightness;
-                getWindow().setAttributes(lp);
-                if (DEBUG) Log.d(TAG, "onScroll().brightnessControl, currentBrightness = " + currentBrightness);
-                int brightnessNormalized = Math.round(currentBrightness * 100);
+                if (currentProgressPercent <= 0) {
+                    playerImpl.getVolumeImageView().setImageDrawable(
+                            AppCompatResources.getDrawable(getApplicationContext(), R.drawable.ic_volume_off_white_72dp)
+                    );
+                } else if (currentProgressPercent < 0.25) {
+                    playerImpl.getVolumeImageView().setImageDrawable(
+                            AppCompatResources.getDrawable(getApplicationContext(), R.drawable.ic_volume_mute_white_72dp)
+                    );
+                } else if (currentProgressPercent < 0.75) {
+                    playerImpl.getVolumeImageView().setImageDrawable(
+                            AppCompatResources.getDrawable(getApplicationContext(), R.drawable.ic_volume_down_white_72dp)
+                    );
+                } else {
+                    playerImpl.getVolumeImageView().setImageDrawable(
+                            AppCompatResources.getDrawable(getApplicationContext(), R.drawable.ic_volume_up_white_72dp)
+                    );
+                }
 
-                final String brightnessText = brightnessUnicode + " " + (brightnessNormalized == 1 ? 0 : brightnessNormalized) + "%";
-                playerImpl.getBrightnessTextView().setText(brightnessText);
+                if (playerImpl.getVolumeRelativeLayout().getVisibility() != View.VISIBLE) {
+                    animateView(playerImpl.getVolumeRelativeLayout(), SCALE_AND_ALPHA, true, 200);
+                }
+                if (playerImpl.getBrightnessRelativeLayout().getVisibility() == View.VISIBLE) {
+                    playerImpl.getBrightnessRelativeLayout().setVisibility(View.GONE);
+                }
+            } else {
+                playerImpl.getBrightnessProgressBar().incrementProgressBy((int) distanceY);
+                float currentProgressPercent =
+                        (float) playerImpl.getBrightnessProgressBar().getProgress() / playerImpl.getMaxGestureLength();
+                WindowManager.LayoutParams layoutParams = getWindow().getAttributes();
+                layoutParams.screenBrightness = currentProgressPercent;
+                getWindow().setAttributes(layoutParams);
+
+                if (DEBUG) Log.d(TAG, "onScroll().brightnessControl, currentBrightness = " + currentProgressPercent);
+
+                if (currentProgressPercent < 0.25) {
+                    playerImpl.getBrightnessImageView().setImageDrawable(
```

Comment:
I think we could improve readibility and be less redudant here (and above) introducing a variable:
```java
// We could initialize at the "else" value, but I think
// this way makes it more readable.
final int iconRes; 
if (currentProgressPercent < 0.25) {
    iconRes = R.drawable.ic_brightness_low_white_72dp;
} else if (currentProgressPercent < 0.75) {
    iconRes = R.drawable.ic_brightness_medium_white_72dp;
} else {
    iconRes = R.drawable.ic_brightness_high_white_72dp;
} 
```

What do you think?

Reply:
Not a bad idea, will do.

Reply:
Hmm, ternary operator is right-associative... :
```
playerImpl.getVolumeImageView().setImageDrawable(
        AppCompatResources.getDrawable(getApplicationContext(),
                currentProgressPercent <= 0
                        ? R.drawable.ic_volume_off_white_72dp
                        : currentProgressPercent < 0.25
                        ? R.drawable.ic_volume_mute_white_72dp
                        : currentProgressPercent < 0.75
                        ? R.drawable.ic_volume_down_white_72dp
                        : R.drawable.ic_volume_up_white_72dp
        )
);
```
Can I use this? :D

Reply:
Definitely hard to read, but if you keep the condition result in the same line, yeah. A variable here can help it again to keep the line shorter and the style closer to what is used currently:

```java
final int resId = currentProgressPercent <= 0 ? R.drawable.ic_volume_off_white_72dp
        : currentProgressPercent < 0.25 ? R.drawable.ic_volume_mute_white_72dp
        : currentProgressPercent < 0.75 ? R.drawable.ic_volume_down_white_72dp
        : R.drawable.ic_volume_up_white_72dp;
```

Reply:
I just wanted to avoid additional variable... but okay, will do.